### PR TITLE
Fix interaction with baseline-circleci

### LINKED ISSUE #70: ConcurrentModificationException applying 1.0.5 before publishing plugins
## What happened?

I'm applying revapi 1.0.5 right after applying `java-library` but before applying other plugins that configure publishing ([in this project](https://github.com/palantir/palantir-java-format/tree/22587955cff3e11687b8d9a11b19f72d06ae42be)):

```gradle
apply plugin: 'java-library'
apply plugin: 'com.palantir.revapi'
apply from: rootProject.file('gradle/publish-jar.gradle')
```

I then get this error:

<details>
<summary>
Stack trace
</summary>

<pre>
Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin [id 'com.palantir.revapi']
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:167)
        at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:136)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyType(DefaultObjectConfigurationAction.java:129)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.access$200(DefaultObjectConfigurationAction.java:38)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction$3.run(DefaultObjectConfigurationAction.java:93)
        at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.execute(DefaultObjectConfigurationAction.java:152)
        at org.gradle.api.internal.project.AbstractPluginAware.apply(AbstractPluginAware.java:49)
        at org.gradle.api.internal.project.ProjectScript.apply(ProjectScript.java:41)
        at org.gradle.api.Script$apply$1.callCurrent(Unknown Source)
        at build_5noics6iamd59kl2acirukei.run(/Users/dsanduleac/code/external/palantir-java-format-2/palantir-java-format-api/build.gradle:2)
        at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:91)
        ... 187 more
Caused by: java.util.ConcurrentModificationException
        at org.gradle.api.internal.collections.FilteredCollection$FilteringIterator.findNext(FilteredCollection.java:121)
        at org.gradle.api.internal.collections.FilteredCollection$FilteringIterator.next(FilteredCollection.java:140)
        at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:479)
        at com.palantir.gradle.revapi.RevapiPlugin.apply(RevapiPlugin.java:60)
        at com.palantir.gradle.revapi.RevapiPlugin.apply(RevapiPlugin.java:34)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
        at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:181)
        at org.gradle.api.internal.plugins.DefaultPluginManager.access$300(DefaultPluginManager.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:276)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:159)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:156)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:49)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:156)
        ... 197 more
</pre>

</details>

## What did you want to happen?

No crash.

### PR DESCRIPTION
**Pull request by CRogers**
## Before this PR
Due to some unknown interaction with `baseline-circleci`, a `ConcurrentModificationException` is thrown. See #70.

## After this PR
==COMMIT_MSG==
Fixes `ConcurrentModificationException` thrown when `com.palantir.baseline` (or more specifically `com.palantir.baseline-circleci`) is applied before `gradle-revapi`.
==COMMIT_MSG==

## Possible downsides?
I don't why this interaction is actually causing it to fail in the first place.

Fixes #70

### CODE REVIEW

File: src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy [hunk 1]
```diff
@@ -393,6 +393,31 @@ class RevapiSpec extends IntegrationSpec {
         assert stderr.contains('method void foo.Foo::setSomeProperty(java.lang.String)')
     }
 
+    def 'does not throw exception when baseline-circleci is applied before this plugin'() {
```
Comment (dansanduleac) [inline src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy:396]: I think the situation in my case was this was failing in a subproject. Baseline was applied at the root project as usual.

Comment (CRogers) [reply src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy:396]: The `baseline-circleci` does stuff with `getTasks()` on `allprojects`, so I think the effect is the same.

Comment (CRogers) [reply src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy:396]: I've made the test use a subproject now

**Review by iamdanfox — APPROVED** [review]

**Comment by CRogers** [comment]
I wonder why bulldozer hasn't done it's job

**Comment by CRogers** [comment]
Manually triggered release

**Pull request merged**